### DISCUSSION TITLE
VSCode - Problem with IntelliSense

### DISCUSSION THREAD

**Discussion by robertochello** [microsoft/vscode-cpptools#11518] [General]
Hi, 
I use Ubuntu 20.04.6 LTS arch64, virtualizzated on Macbook Pro M1 - MacOS Ventura.

I have an unusual problem with VSCode and its IntelliSense.
VSCode is updated to the latest version available for arm64, 1.83.0.

I've installed ROS v0.9.2 extension.

When I write code, autocomplete doesn't show me specific entries, but it shows me all entries.
![Screenshot 2023-10-10 at 17 24 04](https://github.com/microsoft/vscode-cpptools/assets/36032212/ba351c07-4ef1-41b0-a7ed-fc36b98921fe)

This is the `c_cpp_properties.json`:

```
{
  "configurations": [
    {
      "browse": {
        "databaseFilename": "${default}",
        "limitSymbolsToIncludedHeaders": false
      },
      "includePath": [
        "{PATH_TO_WORKSPACE}/devel/include/**",
        "/opt/ros/noetic/include/**",
        "{PATH_TO_WORKSPACE}/src/{NAME_OF_PROJECT}/include/**",
        "/usr/include/**"
      ],
      "name": "ROS",
      "intelliSenseMode": "gcc-arm64",
      "compilerPath": "/usr/bin/gcc",
      "cStandard": "gnu11",
      "cppStandard": "c++14"
    }
  ],
  "version": 4
}
```

Thanks in advance for your help.

**Comment by sean-mcmanus** [comment]
It looks like IntelliSense is not getting configured correctly and it's falling back to the tag parser for completion. It sounds like you're includes are not getting resolved? What is ROS? Is your compiler /usr/bin/gcc? You should remove the system includes like /usr/include/** from your includePath.

**Reply by robertochello** [reply]
[This](https://marketplace.visualstudio.com/items?itemName=ms-iot.vscode-ros) is ROS extension.

If I remove /usr/include/**, there are no changes in IntelliSense.

**Comment by robertochello** [comment]
Up

**Reply by sean-mcmanus** [reply]
@robertochello Can you provide the output of running C/C++: Log Diagnostics after opening some source file? That will have info on how IntelliSense is being configured. You can remove any private project info. Also, are you getting any include not found errors?
